src/dso_api/dynamic_api/remote/views.py

```diff
@@ -141,184 +94,21 @@ def validate(self, serializer, raw_data):
-            )
-
-    def get_headers(self):  # noqa: C901
-        """Collect the headers to submit to the remote service."""
-        client_ip = self.request.META["REMOTE_ADDR"]
-        if isinstance(client_ip, str):
-            client_ip = client_ip.encode("iso-8859-1")
-        forward = self.request.META.get("HTTP_X_FORWARDED_FOR", "")
-        if forward:
-            if isinstance(forward, str):
-                forward = forward.encode("iso-8859-1")
-            forward = b"%b %b" % (forward, client_ip)
-        else:
-            forward = client_ip
-
-        headers = {
-            **self.default_headers,
-            "X-Forwarded-For": forward,
-        }
-
-        # We check if we already have a X-Correlation-ID header
-        x_correlation_id = self.request.META.get("HTTP_X_CORRELATION_ID")
-        if not x_correlation_id:
-            # Otherwise we set it to a part of the X-Unique-ID header
-            # The X-Correlation-ID cannot be longer then 40 characters because MKS suite
-            # cannot handle this. Therefore  we use only part of it.
-            # The X-Unique_ID is defined in Openstack with :
-            # [client_ip]:[client_port]_[bind_ip]:[bind_port]_[timestamp]_[request_counter]:[pid]
-            # But bind_ip and bind_port are always the same. So we can remove them
-            x_unique_id = self.request.META.get("HTTP_X_UNIQUE_ID")
-            if x_unique_id:
-                x_correlation_id = x_unique_id[:14] + x_unique_id[28:]
-        if x_correlation_id:
-            # And if defined pass on to the destination
-            headers["X-Correlation-ID"] = x_correlation_id.encode("iso-8859-1")
-
-        for header in self.headers_passthrough:
-            value = self.request.headers.get(header, "")
-            if not value:
-                continue
-
-            if isinstance(value, str):
-                # Based on DRF's get_authorization_header() logic:
-                # Work around django test client oddness
-                value = value.encode("iso-8859-1")
-            headers[header] = value
-
-        if "kadaster.nl" in self.endpoint_url:
-            headers["X-Api-Key"] = settings.HAAL_CENTRAAL_API_KEY
-            headers["accept"] = "application/hal+json"
-            # Currently for kadaster HaalCentraal only RD (epsg:28992) is supported
-            headers["Accept-Crs"] = "epsg:28992"
-
-        return headers
-
 
 def remote_viewset_factory(
-    endpoint_url, serializer_class, dataset_id, table_id, table_schema
+    endpoint_url, serializer_class, table_schema: DatasetTableSchema
 ) -> Type[RemoteViewSet]:
     """Construct the viewset class that handles the remote serializer."""
+
     return type(
         f"{serializer_class.__name__}Viewset",
         (RemoteViewSet,),
         {
             "__doc__": "Forwarding proxy serializer",
-            "endpoint_url": endpoint_url,
+            "client": clients.make_client(endpoint_url, table_schema),
```

Comment:
I'd use `client_class` instead, and call `make_client` inside the view?

Reply:
As long as clients are stateless, we don't need separate instances.